src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityMiner.java

```diff
@@ -0,0 +1,375 @@
+            tooltip.add(I18n.format("gregtech.machine.miner.description.mv", getType().radius, getType().tick / 20, GTValues.V[getTier() - 1]));
+        else if (getTier() == 3)
+            tooltip.add(I18n.format("gregtech.machine.miner.description.hv", getType().radius, getType().tick / 20, GTValues.V[getTier() - 1]));
+    }
+
+    public boolean drainEnergy() {
+        if (energyContainer.getEnergyStored() >= energyPerTick && !done && !invFull) {
+            energyContainer.removeEnergy(energyPerTick);
+            return true;
+        }
+        return false;
+    }
+
+
+    @Override
+    public void update() {
+        super.update();
+        if (!getWorld().isRemote) {
+            if (!drainEnergy()) {
+                if (!done && x.get() == Integer.MAX_VALUE && y.get() == Integer.MAX_VALUE && z.get() == Integer.MAX_VALUE) {
+                    initPos();
+                }
+                if (invFull && getOffsetTimer() % 5 == 0) {
+                    pushItemsIntoNearbyHandlers(getFrontFacing());
+                    invFull = false;
+                }
+                return;
+            }
+
+            WorldServer world = (WorldServer) this.getWorld();
+
+
+            if (mineY.get() < tempY.get()) {
+                world.destroyBlock(new BlockPos(getPos().getX(), tempY.get(), getPos().getZ()), false);
+                tempY.decrementAndGet();
+                this.pipeY++;
+                writeCustomData(-200, b -> b.writeInt(pipeY));
+                markDirty();
+            }
+
+            if (blockPos.isEmpty()) {
+                blockPos = IMiner.getBlocksToMine(this, x, y, z, startX, startZ, startY, aRadius);
+                iterate = blockPos.iterator();
+            }
+
+
+            if (getOffsetTimer() % type.tick == 0 && !blockPos.isEmpty()) {
+                BlockPos tempPos = (BlockPos) iterate.next();
+                NonNullList<ItemStack> itemStacks = NonNullList.create();
+                IBlockState blockState = this.getWorld().getBlockState(tempPos);
+                if (blockState != Blocks.AIR.getDefaultState()) {
+                    blockState.getBlock().getDrops(itemStacks, world, tempPos, blockState, 0);
+                    if (addItemsToItemHandler(exportItems, true, itemStacks)) {
+                        addItemsToItemHandler(exportItems, false, itemStacks);
+                        world.destroyBlock(tempPos, false);
+                        mineX.set(tempPos.getX());
+                        mineZ.set(tempPos.getZ());
+                        mineY.set(tempPos.getY());
+                        index++;
+                    } else {
+                        invFull = true;
+                    }
+                } else
+                    index++;
+            } else if (blockPos.isEmpty()) {
+                initPos();
+                blockPos = IMiner.getBlocksToMine(this, x, y, z, startX, startZ, startY, aRadius);
+                iterate = blockPos.iterator();
```

Comment:
I started thinking that it was using thread to retrieve the blocks and push into linkedlist while pulling the block from linkedlist into the output slot. But it doesn't seem like it? I still don't know what's optimized